platform/darwin/src/MGLStyle.mm

```diff
@@ -205,12 +206,21 @@ - (void)insertLayer:(MGLStyleLayer *)layer belowLayer:(MGLStyleLayer *)otherLaye
 
 - (void)addSource:(MGLSource *)source
 {
-    self.mapView.mbglMap->addSource(source.mbglSource);
+    // After it is added, the pending source is moved to the mbgl map and is
+    // no longer "pending". The move operation sets it to NULL. However, the
+    // MGLSource's rawSource will still point to the mbgl source and clients
+    // can mutate it via that reference.
+    self.mapView.mbglMap->addSource(source.pendingSource);
```

Comment:
Here’s an idea that still requires indirection but limits the gymnastics to well-worn techniques:

* Remove the `_pendingSource` ivar from MGLSource and add it as an ivar in each concrete subclass instead. All the current calls to `-setPendingSource:` become direct assignments to `_pendingSource`.
* Have each concrete subclass of MGLSource implement an `-addToMapView:` method that takes a reference to an `mbgl::Map` and calls `addSource()` on it, passing in `std::move(_pendingSource)`.

Do you anticipate any other reasons we’d need to get the pending source like that?